server/src/main/java/io/crate/expression/operator/EqOperator.java

```diff
@@ -198,15 +198,24 @@ private static Query booleanShould(String column, DataType<?> type, Collection<?
     private static Query termsAndGenericFilter(Function function, String column, DataType<?> elementType, Collection<?> values, LuceneQueryBuilder.Context context) {
         MappedFieldType fieldType = context.getFieldTypeOrNull(column);
         if (fieldType == null) {
+            if (elementType.id() == ObjectType.ID) {
```

Comment:
Doing smth like 
```
boolQuery = ...
for (array item: values) {
  boolQuery &= refEqObject
}
return boolQuery
```
doesn't make sense for `array(object) = some_array` queries: 
if `some_array` is empty we try to use DocValues which is not supported for object so would be null/generic filter anyway.
If `some_array` is not empty, chain `termsQuery(object type) -> booleanShould(object type) --> fromPrimitive (obj) ` would return null anyway since `StorageSupport.eqQuery` is null for `ObjectType`.